spikeinterface/sortingcomponents/peak_localization.py

```diff
@@ -89,42 +134,92 @@ def _localize_peaks_chunk(segment_index, start_frame, end_frame, worker_ctx):
-    local_peaks.copy()
-    local_peaks['sample_ind'] -= start_frame
+    local_peaks = local_peaks.copy()
+    local_peaks['sample_ind'] -= (start_frame - left_margin)
 
+    
     if method == 'center_of_mass':
-        peak_locations = localize_peaks_center_of_mass(traces, local_peaks, contact_locations, neighbours_mask)
+        peak_locations = localize_peaks_center_of_mass(traces, local_peaks, contact_locations, neighbours_mask, nbefore, nafter)
+    elif method == 'monopolar_triangulation':
+        max_distance_um = worker_ctx['method_kwargs']['max_distance_um']
+        peak_locations = localize_peaks_monopolar_triangulation(traces, local_peaks, contact_locations, neighbours_mask, nbefore, nafter, max_distance_um)
 
     return peak_locations
 
 
-def localize_peaks_center_of_mass(traces, local_peak, contact_locations, neighbours_mask):
-    ndim = contact_locations.shape[1]
-    peak_locations = np.zeros((local_peak.size, ndim), dtype='float64')
+def localize_peaks_center_of_mass(traces, local_peak, contact_locations, neighbours_mask, nbefore, nafter):
+    peak_locations = np.zeros(local_peak.size, dtype=dtype_localize_by_method['center_of_mass'])
 
-    # TODO find something faster
     for i, peak in enumerate(local_peak):
         chan_mask = neighbours_mask[peak['channel_ind'], :]
         chan_inds, = np.nonzero(chan_mask)
+        
+        local_contact_locations = contact_locations[chan_inds, :]
 
-        # TODO find the max between nbefore/nafter
-        amps = traces[peak['sample_ind'], chan_inds]
-        amps = np.abs(amps)
-        com = np.sum(amps[:, np.newaxis] * contact_locations[chan_inds, :], axis=0) / np.sum(amps)
+        wf = traces[peak['sample_ind']-nbefore:peak['sample_ind']+nafter, :][:, chan_inds]
+        wf_ptp = wf.ptp(axis=0)
+        com = np.sum(wf_ptp[:, np.newaxis] * local_contact_locations, axis=0) / np.sum(wf_ptp)
+
+        peak_locations['x'][i] = com[0]
+        peak_locations['y'][i] = com[1]
+
+    return peak_locations
+
+
+
+def localize_peaks_monopolar_triangulation(traces, local_peak, contact_locations, neighbours_mask, nbefore, nafter, max_distance_um):
+    """
+    This method is from Julien Boussard see spikeinterface.toolki.postprocessing.unit_localization
+    """
+    peak_locations = np.zeros(local_peak.size, dtype=dtype_localize_by_method['monopolar_triangulation'])
+
+
+    for i, peak in enumerate(local_peak):
+        chan_mask = neighbours_mask[peak['channel_ind'], :]
+        chan_inds, = np.nonzero(chan_mask)
 
-        peak_locations[i, :] = com
+        local_contact_locations = contact_locations[chan_inds, :]
+
+        # wf is (nsample, nchan) - chann is only nieghboor
+        wf = traces[peak['sample_ind']-nbefore:peak['sample_ind']+nafter, :][:, chan_inds]
+        wf_ptp = wf.ptp(axis=0)
+
+        # constant for initial guess and bounds
+        max_alpha = max(wf_ptp) * max_distance_um
+        initial_alpha = max(wf_ptp) * 20
```

Comment:
why?

Reply:
20 um is the initial guess on the orthogonal plan and so the initial alpha is this one.
Am I wrong ?

Reply:
so you can assume:

```
max_location = location_max_channel
max_ptp = alpha / sqrt((com[0] - max_location[0])**2 + (com[1] - max_location[1])**2 + initial_z_guess**2)
```

So you can estimate initial alpha with:
```
initial_alpha = max_ptp * sqrt((com[0] - max_location[0])**2 + (com[1] - max_location[1])**2 + initial_z_guess**2)
```

makes sense?

Reply:
Yes this is better. I was too lazy to do it.
I try.